src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -67,7 +67,7 @@ private static bool ConvertToIntCore(object obj, out int value)
                 return false;
             }
 
-            if (!int.TryParse(text, NumberStyles.Integer, CultureInfo.CurrentCulture, out var converted))
+            if (!int.TryParse(text, NumberStyles.Number, culture ?? CultureInfo.CurrentCulture, out var converted))
```

Comment:
changed this to number so that it will accept a wider range of stuff like a thousands separator.